### PR TITLE
Add onOpened callback to PopupMenuButton

### LINKED ISSUE #21773: support onShowPopupMenu  on PopupMenuButton
add onShowPopupMenu function on PopupMenuButton,  the  lifecycle of PopupMenu show have "onShow" event

### PR DESCRIPTION
**Pull request by limonadev**
This PR adds a new callback `onOpened` to `PopupMenuButton`, called when the menu appears (normally after tapping the button).

*List which issues are fixed by this PR. You must list at least one issue.*
Fixes #21773 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/popup_menu.dart
Comment (goderbauer) [inline packages/flutter/lib/src/material/popup_menu.dart:953]: fly-by comment: Just use `VoidCallback` for this.

Comment (limonadev) [reply packages/flutter/lib/src/material/popup_menu.dart:953]: The `onCanceled` callback is also a `VoidCallback` but it's using a similar definition:
```dart
typedef PopupMenuCanceled = void Function();
```
So I wanted to follow the same code style, should I use `VoidCallback` anyways?

Comment (gspencergoog) [reply packages/flutter/lib/src/material/popup_menu.dart:953]: Yes, the `PopupMenu` should probably have used `VoidCallback` too.

Comment (gspencergoog) [reply packages/flutter/lib/src/material/popup_menu.dart:953]: One note for posterity: we normally would prefer the present tense for function variables (and I just added that to the style guide, since it was missing), so `onOpen` instead of `onOpened`, but in this case it would be inconsistent with the rest of the API here, so I think you should keep it as `onOpened` instead of `onOpen`. It would be too much churn for too little improvement to deprecate the old names and create new ones just for the sake of consistency.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/lib/src/material/popup_menu.dart
Comment (limonadev) [reply packages/flutter/lib/src/material/popup_menu.dart]: Done! Now the `onOpened` callback is using a `VoidCallback`

**Comment by limonadev** [comment]
@gspencergoog is there anything left for this PR to be approved/merged? Thanks!

**Comment by gspencergoog** [comment]
Yes, there needs to be a second approval.

Also, you might want to sync to the current HEAD and push again: it looks like the bots got a little confused and need to be prodded again.

@goderbauer, since you gave a drive-by, would you want to be the second review on this?

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by gspencergoog** [comment]
@limonadev Actually, it looks like you'll need to sync to the HEAD revision on master and push again.  Something didn't get picked up properly in the continuous integration bots, and it just needs to be restarted. Pushing again will do that.

**Comment by limonadev** [comment]
@gspencergoog done! :smile:

**Pull request merged**
